assets/app/scripts/services/applicationGenerator.js

```diff
@@ -86,17 +73,20 @@ angular.module("openshiftConsole")
         imageStream: scope._generateImageStream(input),
         buildConfig: scope._generateBuildConfig(input, imageSpec, input.labels),
         deploymentConfig: scope._generateDeploymentConfig(input, imageSpec, ports, input.labels),
-        service: scope._generateService(input, input.name, scope._getFirstPort(ports))
+        service: scope._generateService(input, input.name, ports)
       };
       resources.route = scope._generateRoute(input, input.name, resources.service.metadata.name);
       return resources;
     };
 
     scope._generateRoute = function(input, name, serviceName){
+      var route;
+
       if(!input.routing.include) {
```

Comment:
is this set to false when there are no ports (and we will not generate a service)?